prov/gni/src/gnix_nic.c

```diff
@@ -1253,6 +1253,10 @@ int gnix_nic_alloc(struct gnix_fid_domain *domain,
 		GNIX_INFO(FI_LOG_EP_CTRL, "Allocated NIC:%p\n", nic);
 	}
 
+	if (nic) {
+		nic->requires_lock = domain->thread_model != FI_THREAD_COMPLETION;
+	}
```

Comment:
I'm convinced that it isn't correct as well. However, as a first draft of
the threading model code, this should work considering that we'd
officially only support thread-safe and thread-completion. In the event
that thread-completion isn't used, this would default to a stricter
locking model, which shouldn't be erroneous, just less performant. We can
change the domain to default to only one of the two models and that would
be cleaner for now.

I'm not certain that I understand the rest of your statement though.
-- Jim




On 4/28/16, 5:49 PM, "Howard Pritchard" <notifications@github.com> wrote:

>In 
>prov/gni/src/gnix_nic.c
><https://github.com/ofi-cray/libfabric-cray/pull/776#discussion_r61515125>
>:
>> @@ -1253,6 +1253,10 @@ int gnix_nic_alloc(struct gnix_fid_domain
>>*domain,
>>  		GNIX_INFO(FI_LOG_EP_CTRL, "Allocated NIC:%p\n", nic);
>>  	}
>>  
>> +	if (nic) {
>> +		nic->requires_lock = domain->thread_model != FI_THREAD_COMPLETION;
>> +	}
>I'm not convinced this is correct. It could be possible in the current
>nic allocation scheme for a nic to be associated with ep's bound to
>different completion queues. We could change the allocation scheme to
>require a nic to be allocated per endpoint in
> the case of FI_THREAD_COMPLETION being requested.
>—
>You are receiving this because you authored the thread.
>Reply to this email directly or
>view it on GitHub 
><https://github.com/ofi-cray/libfabric-cray/pull/776/files/d4db022618a2830
>eaf0e739835c41c58423cef19#r61515125>
>
>